source/app/libpd.c

```diff
@@ -280,6 +299,17 @@ static void parodus_receive()
 						WalPrint("set cloud-status value as %s\n", status);
 					}
 				}
+				/* CID-273897 Resource leak fix */
+				if (sourceService != NULL)
+				{
+					WAL_FREE(sourceService);
```

Comment:
After WAL_FREE, assigning to NULL is not required. WAL_FREE itself is doing it internally . Change in all places